spec/services/middleware_topology_service_spec.rb

```diff
@@ -35,45 +35,41 @@
                                                            :name                  => "hawkular-command-gateway-war.war",
                                                            :nativeid              => "Local~/deployment=hawkular-command-gateway-war.war")
       expect(subject[:items]).to eq(
-                                     ems_hawkular.id.to_s           => {:name         => ems_hawkular.name,
-                                                                        :status       => "Unknown",
-                                                                        :kind         => "MiddlewareManager",
-                                                                        :display_kind => "Hawkular",
-                                                                        :miq_id       => ems_hawkular.id,
-                                                                        :icon         => "vendor-hawkular",
-                                                                        :id           => ems_hawkular.id.to_s},
+        "MiddlewareManager" + ems_hawkular.id.to_s              => {:name         => ems_hawkular.name,
+                                                                    :status       => "Unknown",
+                                                                    :kind         => "MiddlewareManager",
+                                                                    :display_kind => "Hawkular",
+                                                                    :miq_id       => ems_hawkular.id,
+                                                                    :icon         => "vendor-hawkular"},
 
-                                     middleware_server.ems_ref      => {:name         => middleware_server.name,
-                                                                        :status       => "Unknown",
-                                                                        :kind         => "MiddlewareServer",
-                                                                        :display_kind => "MiddlewareServer",
-                                                                        :miq_id       => middleware_server.id,
-                                                                        :icon         => "vendor-wildfly",
-                                                                        :id           => middleware_server.nativeid},
+        "MiddlewareServer" + middleware_server.id.to_s          => {:name         => middleware_server.name,
```

Comment:
I believe you need to use compressed_id instead of id. I also see the tests are failing most likely due to this problem.